### DISCUSSION TITLE
Save tokens by directing GPT-Pilot to build complex apps on top of an open source back-end platform

### DISCUSSION THREAD

**Discussion by MaxAnfilofyev** [Pythagora-io/gpt-pilot#899] [Ideas]
Most of the more complex projects require similar infrastructure. Instead wasting tokens to rebuild the infrastructure every time, prompt GPT-Pilot to use one of the popular open source back-end platforms that have no limitations on usage to abstract the complexity and repetitiveness required to build a modern backend API from scratch and to build secure apps faster.

[Open source BaaS](https://gist.github.com/PARC6502/ee4db400a05e6eb6d0981bb8cd4e4c1c) 

[open source backend project comparison](https://www.youtube.com/watch?v=EgxG3XTZ5uM)

Common back-end components
- Authentication
- Authorization
- Database
- Admin UI
- File storage
- Functions
- Web hooks
- Chron jobs
- Push notifications
- Emails

**Comment by s0rttega** [comment]
Thanks for this!.
